### DISCUSSION TITLE
How do I exclude the binary from the archive?

### DISCUSSION THREAD

**Discussion by chabad360** [goreleaser/goreleaser#2650] [Q&A] [answered]
Hey, I've been trying to put my binary in a `.app` bundle for MacOS (it's a GUI application), but I want to only include the `.app` folder (with the binary copied into it in a hook) in the archive but not the binary itself. Is there anyway to do this (or a better way to package for MacOS)? 

My `.goreleaser.yml` (I've removed the windows bits otherwise it's a bit large):
```yaml
project_name: resolume-timecode
before:
  hooks:
    - go mod download
builds:
  - id: darwin
    goos:
      - darwin
    goarch:
      - amd64
    env:
      - CGO_ENABLED=1
    flags:
      - -mod=readonly
      - -trimpath
    ldflags:
      - -s -w
      - -X main.version={{.Version}}
      - -X main.Date={{.CommitDate}}
    mod_timestamp: '{{ .CommitTimestamp }}'
    hooks:
      post:
        - upx {{ .Path }}
        - sh macos.sh resolume-timecode images/logo.svg "me.chabad360.resolume-timecode" {{ .Major }} {{ .Minor }}
        - cp {{ .Path }} resolume-timecode.app/Contents/MacOS/
checksum:
  name_template: 'checksums.txt'
changelog:
  sort: asc
  filters:
    exclude:
      - '^chore:'
      - '^ci:'
      - '^docs?:'
      - '^tests?:'
release:
  github:
    owner: chabad360
    name: resolume-timecode
  draft: true
  prerelease: auto
archives:
  - id: darwin
    builds:
      - darwin
    format: zip
    wrap_in_directory: true
    replacements:
      darwin: macOS
    files:
      - LICENSE
      - README.MD
      - resolume-timecode.app

**Comment by caarlos0** [comment, marked as answer]
its not possible right now...

**Reply by chabad360** [reply]
Bummer. Is it possible that GoReleaser could gain the ability to create these kinds of bundles in a future version (`.app` for macOS and even the windows .COFF stuff)?

**Reply by caarlos0** [reply]
there are issues for it, problem on both cases is that we would depend on other stuff (binaries from macos, fuse-something on linux, god-knows-what for windows)...

**Reply by chabad360** [reply]
From what I've seen there are pure go tools for doing that stuff: [for windows](https://github.com/tc-hib/go-winres), for [mac](https://github.com/JackMordaunt/icns), for linux I'm pretty sure it only needs a `.desktop` file for that kind of metadata.

**Reply by caarlos0** [reply]
on windows most people want installers, then you'll need more tools...

on macs, I think the issue is creating the dmg file (which is a filesystem)

for the desktop, if you provide it to goreleaser it should already work, just need to wire it in the nfpm config...

**Reply by chabad360** [reply]
Fair, but even just the basic bits like the `.app` folder structure and windows metadata would be nice, it's rather hard to do otherwise. At the very least an option to create an archive that doesn't necessarily contain the binary would go a long way to making this easier.

**Reply by clobrano** [reply]
It's been a while, but I'd like to add +1 to this.
I'm playing with goreleaser to release kubernetes operators where the release is actually a bundle folder, not the binary. So I'd like to remove the binary from the archive as well.

**Reply by caarlos0** [reply]
@clobrano for the specific case of k8s, I think you can set `archives.meta: true` https://goreleaser.com/customization/archive/

@chabad360 for the app bundles, the proper packaging is a DMG, which has a `replace` option to remove the macOS archives: https://goreleaser.com/customization/dmg/

**Reply by clobrano** [reply]
@caarlos0, `archives.meta: true` looks promising (I don't recall it was there when I commented originally), I'll have a look, thanks :+1:

**Comment by clobrano** [comment]
I'd like to keep this open, unless the feature as been marked as unwanted at all
